Sources/Events/EventsManager.swift

```diff
@@ -143,19 +136,27 @@ actor EventsManager: EventsManagerType {
     }
 
     func flushFeatureEvents(batchSize: Int) async throws -> Int {
-        #if os(iOS) || os(tvOS) || VISION_OS
-        let endBackgroundTask: (() -> Void)?
-        if !self.systemInfo.isAppExtension {
-            endBackgroundTask = await self.beginBackgroundTask(named: "com.revenuecat.flushFeatureEvents")
-        } else {
-            endBackgroundTask = nil
-        }
-        defer {
-            endBackgroundTask?()
+        return try await self.flushFeatureEventsInternal(batchSize: batchSize)
+    }
+
+    nonisolated func flushAllEventsWithBackgroundTask(batchSize: Int) {
```

Comment:
Marking this `nonisolated` probably does allow for multiple operations to be executed at the same time, would that cause issues? E.g. if you open notification center twice it would probably fire `willResignActive` twice, and if one request is still running we could end up uploading the same batch twice, and also removing 2x `n` number of events? Or am I missing something in the implementation?